Adds documentation for migrations failing on timeouts while waiting for index yellow status (#129949)
elastic/kibana: docs/setup/upgrade.asciidoc

[float]
==== Ensure your {es} cluster is healthy

Problems with your {es} cluster can prevent {kib} upgrades from succeeding.

> During the upgrade process, {kib} creates new indices into which updated documents are written. If a cluster is approaching the low watermark, there's a high risk of {kib} not being able to create these. Reading, transforming and writing updated documents can be memory intensive, using more available heap than during routine operation. You need to make sure that there is anough heap available to prevent requests from timing out or throwing errors from circuit breaker exceptions. In addition, you should also ensure that all shards are replicated and assigned.

Reviewer: ```suggestion
During the upgrade process, {kib} creates new indices into which updated documents are written. If a cluster is approaching the low watermark, there's a high risk of {kib} not being able to create these. Reading, transforming and writing updated documents can be memory intensive, using more available heap than during routine operation. You must make sure that enough heap is available to prevent requests from timing out or throwing errors from circuit breaker exceptions. You should also ensure that all shards are replicated and assigned.
```
